src/Microsoft.AspNet.Mvc.TagHelpers/Internal/FileProviderGlobbingDirectory.cs

```diff
@@ -0,0 +1,116 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.AspNet.FileProviders;
+using Microsoft.Framework.FileSystemGlobbing.Abstractions;
+
+namespace Microsoft.AspNet.Mvc.TagHelpers.Internal
+{
+    public class FileProviderGlobbingDirectory : DirectoryInfoBase
+    {
+        private readonly IFileProvider _fileProvider;
+        private readonly IFileInfo _fileInfo;
+        private readonly FileProviderGlobbingDirectory _parent;
+        private readonly bool _isRoot;
+
+        public FileProviderGlobbingDirectory(
+            [NotNull] IFileProvider fileProvider,
+            IFileInfo fileInfo,
+            FileProviderGlobbingDirectory parent)
+        {
+            _fileProvider = fileProvider;
+            _fileInfo = fileInfo;
+            _parent = parent;
+
+            if (_fileInfo == null)
+            {
+                // We're the root of the directory tree
+                RelativePath = string.Empty;
+                _isRoot = true;
+            }
+            else if (!string.IsNullOrEmpty(parent?.RelativePath))
+            {
+                // We have a parent and they have a relative path so concat that with my name
+                RelativePath = _parent.RelativePath + Path.DirectorySeparatorChar + _fileInfo.Name;
```

Comment:
IMO, it might be ok to use "/". We had a similar issue where we were constructing paths using the pathing APIs and we ended up with slashes being different since the APIs internally use `Path.DirectorySeparatorChar`. (https://github.com/aspnet/Mvc/pull/1996#discussion_r24517836). I have a follow up item and the right resolution for it would be to use "/" consistently.